### PR TITLE
Fitness demo, initial version

### PR DESCRIPTION
**Pull request by vlidholt**

### CODE REVIEW

File: examples/material_gallery/lib/demo/fitness_demo.dart [hunk 1]
```diff
@@ -0,0 +1,486 @@
...
+      padding: new EdgeDims.only(bottom: 10.0),
+      child: new SizedBox(
+        width: 300.0,
+        height: 72.0,
+        child: new RaisedButton (
+          onPressed: onPressed,
+          color: color,
+          child: new Text(
+            text,
+            style: new TextStyle(color: Colors.white, fontSize: 18.0)
```
Comment (Hixie) [inline examples/material_gallery/lib/demo/fitness_demo.dart]: raised buttons should get the right text style automatically

Comment (Hixie) [inline examples/material_gallery/lib/demo/fitness_demo.dart]: (if they don't, use a ButtonTheme widget to configure them)

File: examples/material_gallery/lib/demo/fitness_demo.dart [hunk 2]
```diff
@@ -0,0 +1,486 @@
...
+
+import 'dart:async';
+import 'dart:math' as math;
+import 'dart:ui' as ui;
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_sprites/flutter_sprites.dart';
+
+ImageMap _images;
+SpriteSheet _sprites;
```
Comment (Hixie) [inline examples/material_gallery/lib/demo/fitness_demo.dart]: These should not be global, or, the initialisation should be global. See #1591.

File: examples/material_gallery/lib/demo/fitness_demo.dart (hunk 1 above)
Comment (vlidholt) [inline examples/material_gallery/lib/demo/fitness_demo.dart]: Well, that currently doesn't happen. Will file a bug and leave this in the code for now

File: examples/material_gallery/lib/demo/fitness_demo.dart [hunk 3]
```diff
@@ -0,0 +1,486 @@
...
+  }
+
+  void initState() {
+    super.initState();
+
+    AssetBundle bundle = DefaultAssetBundle.of(context);
+    _loadAssets(bundle).then((_) {
+      setState(() {
+        assetsLoaded = true;
+        workoutAnimation = new _WorkoutAnimationNode(onValueUpdate: (){
```
Comment (Hixie) [inline examples/material_gallery/lib/demo/fitness_demo.dart]: This rebuilds the entire page every time the animation ticks, which seems like an excessive amount of work (most of the page is static). Could this be refactored into several subcomponents with just the animating bits being stateful?

File: examples/material_gallery/lib/demo/fitness_demo.dart [hunk 4]
```diff
@@ -0,0 +1,486 @@
...
+          setState((){});
+        });
```
Comment (Hixie) [inline examples/material_gallery/lib/demo/fitness_demo.dart]: stylistic nits:
* put a space between the `()` and the `{`.
* empty code blocks should have a space inside them, as in `{ }`, to distinguish them from empty maps.
* empty `setState` calls should have a comment inside their block explaining what state changed

File: examples/material_gallery/lib/demo/fitness_demo.dart [hunk 5]
```diff
@@ -0,0 +1,486 @@
...
+
+    return new Scaffold(
+      toolBar: new ToolBar(
+        center: new Text("Fitness")
+      ),
+      body: new Material(
+        child: new Column(
+          justifyContent: FlexJustifyContent.center,
+          children: <Widget>[
+            _createWorkoutAnimation(),
```
Comment (Hixie) [inline examples/material_gallery/lib/demo/fitness_demo.dart]: Please inline builder functions into the build() method so it's more obvious how big the build is.
As a general rule, we should only factor out subparts of build functions if they are repeated.
Even better, though, as noted above, would be to factor some of these parts out into separate stateless components.

File: examples/material_gallery/lib/demo/fitness_demo.dart (hunk 1 above)
Comment (Hixie) [inline examples/material_gallery/lib/demo/fitness_demo.dart]: what happens?

**Comment by Hixie** [comment]
This would be a good thing to add to our benchmarks.

File: examples/material_gallery/lib/demo/fitness_demo.dart (hunk 1 above)
Comment (vlidholt) [inline examples/material_gallery/lib/demo/fitness_demo.dart]: The color is not set by changing the color of the button and the ButtonTheme doesn't provide enough options to make the button look right in this context.

File: examples/material_gallery/lib/demo/fitness_demo.dart (hunk 2 above)
Comment (vlidholt) [inline examples/material_gallery/lib/demo/fitness_demo.dart]: Replied to the issue.

File: examples/material_gallery/lib/demo/fitness_demo.dart (hunk 3 above)
Comment (vlidholt) [inline examples/material_gallery/lib/demo/fitness_demo.dart]: Will fix

File: examples/material_gallery/lib/demo/fitness_demo.dart (hunk 1 above)
Comment (Hixie) [inline examples/material_gallery/lib/demo/fitness_demo.dart]: Can you show a screenshot of what you mean? As far as I can tell RaisedButton does all the things that the spec says it should do...

File: examples/material_gallery/lib/demo/fitness_demo.dart (hunk 3 above)
Comment (vlidholt) [inline examples/material_gallery/lib/demo/fitness_demo.dart]: To clarify, the page isn't being rebuilt on every animation tick, only when the state changes (i.e. when the labels needs to be updated). Are you proposing that the labels should be stateful components?

File: examples/material_gallery/lib/demo/fitness_demo.dart [hunk 6]
```diff
@@ -0,0 +1,490 @@
...
+                _createInfoPanelCell("image/flash_on", "${kcal}", "KCAL")
+              ]
+            )
+          ),
+          new Padding(
+            padding: new EdgeDims.only(bottom: 10.0),
+            child: new SizedBox(
+              width: 300.0,
+              height: 72.0,
+              child: new RaisedButton (
```
Comment (Hixie) [inline examples/material_gallery/lib/demo/fitness_demo.dart]: (trivial nit: extra space before `(`)

**Comment by Hixie** [comment]
lgtm

**Pull request merged**